From a2ea8da840394f48ff5caa2152f49442f4051584 Mon Sep 17 00:00:00 2001
From: Duncan Bellamy <dunk@denkimushi.com>
Date: Thu, 25 Feb 2021 12:05:56 +0000
Subject: [PATCH] src/global/signal_handler.h: fix preprocessor logic for
 alpine Fixes: https://tracker.ceph.com/issues/49494 Co-authored-by: Kefu Chai
 <tchaikov@gmail.com> Signed-off-by: Duncan Bellamy <dunk@denkimushi.com>

(cherry picked from commit 387f33e7265825ae4cdd3580289f5154e250b05b)
---
 src/global/signal_handler.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/global/signal_handler.h b/src/global/signal_handler.h
index c101b2e28733c..166c1bb206697 100644
--- a/src/global/signal_handler.h
+++ b/src/global/signal_handler.h
@@ -20,9 +20,9 @@
 
 typedef void (*signal_handler_t)(int);
 
-#ifdef HAVE_SIGDESCR_NP
+#if defined(HAVE_SIGDESCR_NP)
 # define sig_str(signum) sigdescr_np(signum)
-#elif HAVE_REENTRANT_STRSIGNAL
+#elif defined(HAVE_REENTRANT_STRSIGNAL)
 # define sig_str(signum) strsignal(signum)
 #else
 # define sig_str(signum) sys_siglist[signum]
